test/int/federated_auth_test.go

```diff
@@ -150,58 +150,14 @@ var _ = Describe("AtlasFederatedAuth test", Label("AtlasFederatedAuth", "federat
 			fedAuth.Spec.DomainRestrictionEnabled = &originalConnectedOrgConfig.DomainRestrictionEnabled
 			fedAuth.Spec.SSODebugEnabled = originalIdp.SsoDebugEnabled
 			fedAuth.Spec.PostAuthRoleGrants = originalConnectedOrgConfig.GetPostAuthRoleGrants()
-			fedAuth.Spec.RoleMappings = nil
-
-			if len(originalConnectedOrgConfig.GetRoleMappings()) > 0 {
-				GinkgoWriter.Println("HAS ROLE MAPPINGS", len(originalConnectedOrgConfig.GetRoleMappings()), originalConnectedOrgConfig.GetRoleMappings()[0])
-				roles := make([]akov2.RoleMapping, len(originalConnectedOrgConfig.GetRoleMappings()))
-
-				for _, roleMapping := range originalConnectedOrgConfig.GetRoleMappings() {
-					assignments := make([]akov2.RoleAssignment, len(roleMapping.GetRoleAssignments()))
-					for _, roleAssignment := range roleMapping.GetRoleAssignments() {
-						var projectName string
-
-						if pID, ok := roleAssignment.GetGroupIdOk(); ok {
-							project, _, err := atlasClient.ProjectsApi.GetProject(ctx, *pID).Execute()
-							Expect(err).ToNot(HaveOccurred())
-							Expect(project).NotTo(BeNil())
-
-							projectName = project.GetName()
-						}
-
-						assignments = append(
-							assignments,
-							akov2.RoleAssignment{
-								ProjectName: projectName,
-								Role:        roleAssignment.GetRole(),
-							},
-						)
-					}
-
-					roles = append(
-						roles,
-						akov2.RoleMapping{
-							ExternalGroupName: roleMapping.GetExternalGroupName(),
-							RoleAssignments:   assignments,
-						},
-					)
-				}
 
-				fedAuth.Spec.RoleMappings = roles
-			} else {
-				roleMappings, _, err := atlasClient.FederatedAuthenticationApi.
-					ListRoleMappings(ctx, originalFederationSettings.GetId(), orgID).
-					Execute()
-				Expect(err).ToNot(HaveOccurred())
-
-				for _, roleMapping := range roleMappings.GetResults() {
-					GinkgoWriter.Println("DELETING ROLE MAPPING", roleMapping.GetId())
-					_, err := atlasClient.FederatedAuthenticationApi.
-						DeleteRoleMapping(ctx, originalFederationSettings.GetId(), roleMapping.GetId(), orgID).
-						Execute()
-					Expect(err).ToNot(HaveOccurred())
+			roleMappings := make([]akov2.RoleMapping, 0, len(fedAuth.Spec.RoleMappings))
+			for _, roleMap := range fedAuth.Spec.RoleMappings {
+				if roleMap.ExternalGroupName != "ako_team" {
+					roleMappings = append(roleMappings, roleMap)
 				}
 			}
+			fedAuth.Spec.RoleMappings = roleMappings
```

Comment:
I replaced the clean-up logic to reflect the operator ability to add/edit/remove role mappings

Reply:
can you please add a comment that makes it clear that this is changing existing role mappings?

Reply:
@s-urbaniak this PR just adjusts the broken part of the test to better reflect what we want to achieve with it.
A quick look into the full test case shows:
* Retrieve of current config
* Build a FedAuth CR from the current config and add a new role mapping
* Successfully reconcile it
* Modify the CR to remove added role mapping (for test)
* Successfully reconcile it

Reply:
@helderjs sorry for being imprecise, I meant commenting in the code not here 🙏

Reply:
@s-urbaniak done